common/python/utility/hex_utils.py

```diff
@@ -29,6 +29,17 @@ def hex_to_utf8(binary):
     return binascii.hexlify(binary).decode("UTF-8")
 
 
+def hex_to_byte_array(hex_str):
+    """
+    Convert a hex string (i.e., a string of characters with values
+    between '0'-'9', 'A'-'F') to an array of bytes
+
+    @param hex_str - hex string to be converted to bytearray
+    @returns - array of bytes
+    """
+    return bytearray(binascii.unhexlify(hex_str))
```

Comment:
Add try except for None & non hex. Will throw binascii.Error for non hex and TypeError for None. Should be caught and appropriate values returned from here.

Reply:
Done